interfaces/builtin/pulseaudio.go

```diff
@@ -23,27 +23,157 @@ import (
 	"github.com/snapcore/snapd/interfaces"
 )
 
-const pulseaudioConnectedPlugAppArmor = `
-/etc/pulse/ r,
-/etc/pulse/* r,
+var pulseaudioConnectedPlugAppArmor = []byte(`
+capability dac_override,
+
+/etc/machine-id r,
+
 /{run,dev}/shm/pulse-shm-* rwk,
-owner /{,var/}run/user/*/pulse/ r,
+
+# FIXME Desktop-only
+owner /{,var/}run/user/*/pulse/ rwk,
 owner /{,var/}run/user/*/pulse/native rwk,
-`
 
-const pulseaudioConnectedPlugSecComp = `
+# Running as system instance
+owner /{,var/}run/pulse rwk,
+owner /{,var/}run/pulse/native rwk,
+`)
+
+var pulseaudioConnectedPlugAppArmorDesktop = []byte(`
+# Only on desktop we need access to /etc/pulse for any PulseAudio client
+# to read available client side configuration settings. On an Ubuntu Core
+# device those things will be stored inside the snap directory.
+/etc/pulse/ r,
+/etc/pulse/* r,
+`)
+
+var pulseaudioConnectedPlugSecComp = []byte(`
 setsockopt
+getsockopt
 connect
 sendto
-`
-
-// NewPulseAudioInterface returns a new "pulseaudio" interface.
-func NewPulseAudioInterface() interfaces.Interface {
-	return &commonInterface{
-		name: "pulseaudio",
-		connectedPlugAppArmor: pulseaudioConnectedPlugAppArmor,
-		connectedPlugSecComp:  pulseaudioConnectedPlugSecComp,
-		reservedForOS:         true,
```

Comment:
Removing this will allow anyone to upload a snap with the pulseaudio slot. Can we please make sure that is flagging the snap for manual review in the store?